action/protocol/poll/util.go

```diff
@@ -168,16 +168,19 @@ func setCandidates(
 		return errors.New("put poll result height should be epoch start height")
 	}
 	loadCandidatesLegacy := featureCtx.LoadCandidatesLegacy(height)
+	accountCreationOpts := []state.AccountCreationOption{state.DelegateCandidateOption()}
```

Comment:
for safety `DelegateCandidateOption()` should be restricted to be used only when creating a delegate account
upon loading a delegate account, should not use this option (and instead should check loaded account has `act.isDelegate == true`)

Reply:
is it a bug introduced in this PR? how could it work in the past?